infrastructure/smart-contracts/contracts/upgrades/referee/Referee10.sol

```diff
@@ -796,20 +802,21 @@ contract Referee10 is Initializable, AccessControlEnumerableUpgradeable {
         }
     }
 
-    function unstakeKeys(address pool, address staker, uint256[] memory keyIds) external onlyPoolFactory {
+    function unstakeKeys(address pool, address staker, uint256 keyAmount) external onlyPoolFactory {
         require(stakingEnabled, "52");
 
-        uint256 keysLength = keyIds.length;
         NodeLicense nodeLicenseContract = NodeLicense(nodeLicenseAddress);
 
-        for (uint256 i = 0; i < keysLength; i++) {
-            uint256 keyId = keyIds[i];
-            require(assignedKeyToPool[keyId] == pool, "47");
-            require(nodeLicenseContract.ownerOf(keyId) == staker, "48");
```

Comment:
This function is no longer sanitizing input. It makes no balance checks before subtracting the key amounts.